flink-runtime-web/web-dashboard/src/app/pages/job/checkpoints/job-checkpoints.component.html

```diff
@@ -193,15 +193,74 @@
+                [nzActive]="moreDetailsPanel.active"
+                [nzDisabled]="moreDetailsPanel.disabled">
+            <nz-table
+                    *ngIf="checkPointStats"
+                    class="no-border small"
+                    [nzData]="(checkPointStats&&checkPointStats['summary'])?['']:[]"
+                    [nzSize]="'small'"
+                    [nzFrontPagination]="false"
+                    [nzShowPagination]="false">
+                <thead>
+                <tr>
+                    <th></th>
+                    <th><strong>End to End Duration</strong></th>
+                    <th><strong>Checkpointed Data Size</strong></th>
+                    <th><strong>Processed (persisted) in-flight data</strong></th>
+                </tr>
+                </thead>
+                <tbody>
+                <ng-container *ngIf="checkPointStats['summary']">
+                <tr>
+                    <td><strong>50% percentile</strong></td>
+                    <td>{{ checkPointStats['summary']['end_to_end_duration']['p50'] | humanizeDuration}}</td>
+                    <td>{{ checkPointStats['summary']['state_size']['p50'] | humanizeBytes }}</td>
+                    <td>{{ checkPointStats['summary']['processed_data']['p50'] | humanizeBytes }} ({{ checkPointStats['summary']['persisted_data']['p50'] | humanizeBytes }})</td>
+                </tr>
+                <tr>
+                    <td><strong>90% percentile</strong></td>
+                    <td>{{ checkPointStats['summary']['end_to_end_duration']['p90'] | humanizeDuration}}</td>
+                    <td>{{ checkPointStats['summary']['state_size']['p90'] | humanizeBytes }}</td>
+                    <td>{{ checkPointStats['summary']['processed_data']['p90'] | humanizeBytes }} ({{ checkPointStats['summary']['persisted_data']['p90'] | humanizeBytes }})</td>
+                </tr>
+                <tr>
+                    <td><strong>99% percentile</strong></td>
+                    <td>{{ checkPointStats['summary']['end_to_end_duration']['p99'] | humanizeDuration}}</td>
+                    <td>{{ checkPointStats['summary']['state_size']['p99'] | humanizeBytes }}</td>
+                    <td>{{ checkPointStats['summary']['processed_data']['p99'] | humanizeBytes }} ({{ checkPointStats['summary']['persisted_data']['p99'] | humanizeBytes }})</td>
+                </tr>
+                <tr>
+                    <td><strong>99% percentile</strong></td>
+                    <td>{{ checkPointStats['summary']['end_to_end_duration']['p99'] | humanizeDuration}}</td>
+                    <td>{{ checkPointStats['summary']['state_size']['p99'] | humanizeBytes }}</td>
+                    <td>{{ checkPointStats['summary']['processed_data']['p99'] | humanizeBytes }} ({{ checkPointStats['summary']['persisted_data']['p99'] | humanizeBytes }})</td>
+                </tr>
+                <tr>
+                    <td><strong>99.9% percentile</strong></td>
```

Comment:
I wonder will `99.9%` really mater here. Since the number of tasks of checkpoint cannot be too large in most cases. The difference of `99%` between `99.9%` might only been seen when we have tasks more than 1000.

Reply:
Thanks for reviewing!

The aggregation here is over time, not over tasks. Each data point is a checkpoint, without details per-subtask (subtask details are aggregated upon finalization using sum/max).

And the "time" is currently limited to 10K checkpoints (sliding window), so 1000 can be reached
(10K limit is [set](https://github.com/apache/flink/pull/16447/files#diff-8680a6a315910794f0486c3d4234090317573c5335288b28b812f7d03c6e1b84R30) when creating the histogram).

Reply:
I think we need at least 10k checkpoints to show the difference between the `99.9%` and the `max`. If we make the checkpoint interval as 3 minutes, we need about `3 * 10000 / 60 / 24 = 20.8` days. 
If we make the checkpoint interval as 10 seconds, we need about 1.2 days. 
This might only make sense if we have really short checkpoint interval.

Reply:
20.8 days doesn't sound like a very long period to me:)

And for the ChangelogBackend, the target interval can be around 1s, so we'd need around 3hrs, which sounds reasonable even for tests.

Reply:
Okay, I see.